packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {
+    ShopperBaskets,
+    ShopperBasketsTypes,
+    ShopperCustomers,
+    ShopperCustomersTypes
+} from 'commerce-sdk-isomorphic'
+import {ApiClients, Argument, CacheUpdate, CacheUpdateMatrix, MergedOptions} from '../types'
+import {and, matchesApiConfig, matchesPath, pick} from '../utils'
+
+type Client = ApiClients['shopperBaskets']
+/** Data returned by every Shopper Baskets endpoint (except `deleteBasket`) */
+type Basket = ShopperBasketsTypes.Basket
+/** Data returned by `getCustomerBaskets` */
+type CustomerBasketsResult = ShopperCustomersTypes.BasketsResult
+/** Parameters that get passed around, includes client config and possible parameters from other endpoints */
+type BasketParameters = MergedOptions<Client, Argument<Client['getBasket']>>['parameters']
+/** Parameters that we actually send to the API for `getBasket` */
+type GetBasketParameters = Argument<ShopperBaskets<{shortCode: string}>['getBasket']>['parameters']
+/** Parameters that we actually send to the API for `getCustomerBaskets` */
+type GetCustomerBasketsParameters = Argument<
+    ShopperCustomers<{shortCode: string}>['getCustomerBaskets']
+>['parameters']
+
+// Path helpers to avoid typos!
+const getBasePath = (parameters: {organizationId: string}) => [
+    '/organizations/',
+    parameters.organizationId
+]
+const getBasketPath = (parameters: GetBasketParameters) => [
+    ...getBasePath(parameters),
+    '/baskets/',
+    parameters.basketId
+]
+const getCustomerBasketsPath = (parameters: GetCustomerBasketsParameters) => [
+    ...getBasePath(parameters),
+    '/customers/',
+    parameters.customerId,
+    '/baskets' // No trailing / as it's an aggregate endpoint
+]
+
+const updateBasketQuery = (
+    customerId: string | null,
+    parameters: BasketParameters,
+    newBasket: Basket
+): CacheUpdate => {
+    // The `parameters` received includes the client config and parameters from other endpoints
+    // so we need to exclude unwanted parameters in the query key
+    const getBasketParameters: GetBasketParameters = pick(parameters, [
+        'basketId',
+        'locale',
+        'organizationId',
+        'siteId'
+    ])
+    const basketUpdate = {
+        queryKey: [...getBasketPath(parameters), getBasketParameters] as const
+    }
+
+    // We can only update customer baskets if we have a customer!
+    if (!customerId) return {update: [basketUpdate]}
```

Comment:
In which case the customerId will be null? We only update the cache in onSuccess callback, and the customerId is passed by calling useCustomerId, which does to be defined at all time. ( No API is proceeded til authentication is successful in which customer_id is defined)

Reply:
The return type from `useCustomerId` is `string | null`, so we need this check. (If the return type is incorrect, that should be fixed in a follow up story.)